### PR TITLE
GDK101 support

### PR DESCRIPTION
**Pull request by Szewcson**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4703

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by Szewcson** [comment]
Can I change it to new somehow?

**Comment by Szewcson** [comment]
I can't make CI pass the image.  Can somebody help me with that?

File: components/sensor/gdk101.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
+FTLab GDK101 Gamma Radiation Sensor Module
+==========================================
+
+.. seo::
+    :description: Instructions for setting up GDK101 Gamma Radiation Sensor Module
+    :image: gdk101.jpg
+    :keywords: gdk101
+
+The **GDK101** sensor platform allows you to use your GDKA101 sensor
+(`AliExpress`_, `Amazon`_, `DesertCart`_, `RadonShop`_, `FTLab`_) to monitor radiation levels at your place.
```
Comment (jesserockz) [inline components/sensor/gdk101.rst]: Most of these show out of stock so lets just remove them all.
```suggestion
to monitor radiation levels at your place.
```

File: components/sensor/gdk101.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    :align: center
+    :width: 50.0%
+
+    GDK101 Gamma Radiation Sensor Module.
+
+.. _AliExpress: https://pl.aliexpress.com/i/1005001299979570.html
+.. _Amazon: https://www.amazon.com/GDK101-Radiation-Sensor-Module-Aduino/product-reviews/B01I4RZWA8
+.. _DesertCart: https://www.desertcart.in/products/98880150-gdk-101-gamma-radiation-sensor-module-aduino-sensor-module
+.. _RadonShop: https://www.radonshop.com/ftlab-gdk101-gamma-radiation-sensor-module-arduino
+.. _FTLab: http://allsmartlab.com/eng/294-2/
```
Comment (jesserockz) [inline components/sensor/gdk101.rst]:
```suggestion
```

File: components/sensor/gdk101.rst [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    i2c:
+      sda: GPIO21
+      scl: GPIO22
+      scan: True
+      id: bus_a
+      frequency: 100kHz
```
Comment (jesserockz) [inline components/sensor/gdk101.rst]: We dont add dependent components in the example as users need to set them up themselves and read that documentation.
```suggestion
```

File: components/sensor/gdk101.rst [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        status:
+          name: "GDK101 Status"
+        version:
+          name: "GDK101 FW Version"
+        measurement_duration: 
+          name: "GDK101 Measuring Time"
+    binary_sensor:
+      - platform: gdk101
+        vibrations:
+          name: "GDK101 Vibrations"
```
Comment (jesserockz) [inline components/sensor/gdk101.rst]: Each of these should go in the relevant section below please

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sensor/gdk101.rst (hunk 1 above)
Comment (Szewcson) [reply components/sensor/gdk101.rst]: done

File: components/sensor/gdk101.rst (hunk 2 above)
Comment (Szewcson) [reply components/sensor/gdk101.rst]: done

File: components/sensor/gdk101.rst (hunk 3 above)
Comment (Szewcson) [reply components/sensor/gdk101.rst]: done

File: components/sensor/gdk101.rst (hunk 4 above)
Comment (Szewcson) [reply components/sensor/gdk101.rst]: Done

**Review by jesserockz — APPROVED** [review]

**Pull request merged**